addons/mrp/tests/test_stock_report.py

```diff
@@ -513,3 +513,107 @@ def test_mo_overview_same_component(self):
+        for and used in the BoM cost as they should.
+        """
+        # Create a color variant, which will be used to create a Product
+        attribute_color = self.env['product.attribute'].create({'name': 'Color'})
+        value_black, value_white = self.env['product.attribute.value'].create([
+            {
+                'name': 'Black',
+                'attribute_id': attribute_color.id,
+            },
+            {
+                'name': 'White',
+                'attribute_id': attribute_color.id,
+            },
+        ])
+        # Create 3 product templates, one for the variants, to check to not all variants are used
+        # to compute the cost of the MO, another one to make sure the 'missing components' are still
+        # taken into account (they are the product that were removed from the MO but are still present
+        # in the BoM), and the last one to make sure that kits are still taken in as well.
+        product_variants, missing_product, kit_product = self.env['product.template'].create([
+            {
+                'name': 'Variant Product',
+                'type': 'consu',
+                'attribute_line_ids': [Command.create(
+                    {
+                        'attribute_id': attribute_color.id,
+                        'value_ids': [
+                            Command.link(value_black.id),
+                            Command.link(value_white.id),
+                        ],
+                    },
+                )],
+            },
+            {
+                'name': 'Missing Component',
+                'type': 'consu',
+                'standard_price': 40,
+            },
+            {
+                'name': 'Kit Product',
+                'type': 'consu',
+                'standard_price': 60,
+            },
+        ])
+
+        variant_black = product_variants.product_variant_ids[0]
+        variant_white = product_variants.product_variant_ids[1]
+        variant_black.standard_price = 50
+        variant_white.standard_price = 30
+        bom_normal, _ = self.env['mrp.bom'].create([
+            {
+                'product_tmpl_id': product_variants.id,
+                'type': 'normal',
+                'bom_line_ids': [Command.create(
+                {
+                    'product_id': missing_product.product_variant_id.id,
+                    'product_qty': 1,
+                }),
+                (0, 0,
+                {
+                    'product_id': kit_product.product_variant_id.id,
+                    'product_qty': 1,
+                })],
+            },
+            {
+                'product_tmpl_id': kit_product.id,
+                'type': 'phantom',
+                'bom_line_ids': [Command.create(
+                {
+                    'product_id': missing_product.product_variant_id.id,
+                    'product_qty': 1,
+                })]
```

Comment:
```suggestion
                'bom_line_ids': [Command.create({
                    'product_id': missing_product.product_variant_id.id,
                    'product_qty': 1,
                })]
```
General note: Keep the extra spaces for only a `{` to a minimum, check what's done in other tests. Of course readability is paramount, but too much empty spaces everywhere doesn't help either :smile: